src/rebar_prv_dialyzer.erl

```diff
@@ -382,6 +384,29 @@ run_dialyzer(State, Opts) ->
             {0, State}
     end.
 
+filter_warnings([], Warnings) ->
+    Warnings;
+filter_warnings(Ignores, Warnings) ->
+    lists:filter(fun(Warning) ->
+        not lists:any(fun (Ignore) -> match_warning(Ignore, Warning) end, Ignores)
+    end, Warnings).
+
+match_warning({MatchTag, {MatchPath, MatchLine}, MatchMsg}, {Tag, {Path, Line}, Msg}) when MatchPath =/= '_' ->
+    lists:suffix(MatchPath, Path) andalso match_term([MatchTag, MatchLine, MatchMsg], [Tag, Line, Msg]);
+match_warning({_, _, _} = Match, {_, {_, _}, _} = Warning) ->
+    match_term(Match, Warning).
+
+match_term(Term, Term) ->
+    true;
+match_term('_', _Term) ->
+    true;
+match_term(Term1, Term2) when is_list(Term1), is_list(Term2), length(Term1) =:= length(Term2) ->
```

Comment:
The tail of a list in a `Match` could be `'_'` which is not supported here. List matching should be recursive, something like:

```erlang
match_term([MatchHead | MatchRest], [Head | Rest]) ->
    match_term(MatchHead, Head) andalso match_term(MatchRest, Rest).
```